kong/plugins/aws-lambda/handler.lua

```diff
@@ -62,6 +65,21 @@ local function get_now()
   return ngx_now() * 1000 -- time is kept in seconds with millisecond resolution.
 end
 
+local function validate_http_status_code(status_code)
+  if not status_code then
+    return false
+  end
+  local status_code_str = tostring(status_code)
```

Comment:
This seems to be slightly unnecessary. I think if the `status_code` is already a number, maybe we do not need to run the regex below?

What about:

```
if type(status_code) == "string" then
    status_code = tonumber(status_code)
end
```

And later below, we can do:

```
if status_code >= 100 and status_code <= 599 then
  return status_code
end

return nil
```

This should be a lot faster than a regex based solution as well.

Reply:
OK I've changed the code to compare numbers directly.

And we need to do additional check if `status_code` is nil after the `tonumber` conversion because nil cannot compare with numbers.

Fixed